main.tf

```diff
@@ -84,22 +50,119 @@ module "password_lambda" {
+      resources = ["*"]
+    }
+    "b" = {
+      effect = "Allow"
+      actions = [
+        "ssm:DescribeParameters",
+        "ssm:GetParameter",
+        "ssm:GetParameters",
+        "ssm:GetParametersByPath",
+        "ssm:PutParameter"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:ssm:${var.region}:${data.aws_caller_identity.current.account_id}:parameter/log-exporter-last-export/*"]
+    }
+    "c" = {
+      effect = "Allow"
+      actions = [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:logs:${var.region}:${data.aws_caller_identity.current.account_id}:*"]
+    }
+    "d" = {
+      effect = "Allow"
+      actions = [
+        "s3:PutObject",
+        "s3:PutObjectACL"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.cloudwatch_logs_export_bucket}"]
+    }
+    "e" = {
+      effect = "Allow"
+      actions = [
+        "s3:PutBucketAcl",
+        "s3:GetBucketAcl"
+      ]
+      resources = ["arn:${data.aws_partition.current.partition}:s3:::${var.cloudwatch_logs_export_bucket}"]
+    }
+  } : null
+
+  dynamic_environment_variables = var.which_lambda_function == "password_rotation" ? {
+    users        = join(",", var.users)
+    instance_ids = join(",", var.instance_ids)
+    } : var.which_lambda_function == "log_exportation" ? {
+    cloudwatch_logs_export_bucket = var.cloudwatch_logs_export_bucket
+  } : {}
+  static_env_vars = {
+    slack_webhook_url          = var.slack_webhook_url
+    slack_notification_enabled = var.slack_notification_enabled
+  }
+}
+
+
+module "lambda" {
+  source        = "git::https://github.com/terraform-aws-modules/terraform-aws-lambda.git?ref=v6.2.0"
+  function_name = join("-", [var.name_prefix, var.which_lambda_function, var.random_id])
+  description   = var.description
+  handler       = var.handler
+  runtime       = var.runtime
+  timeout       = var.timeout
+  publish       = true
+  allowed_triggers = {
+    event_bridge_trigger = {
+      principal  = "events.amazonaws.com"
+      source_arn = aws_cloudwatch_event_rule.cron_eventbridge_rule.arn
+    }
   }
-  source_path = "${path.module}/fixtures/functions/password-rotation/lambda_function.py"
+
+  environment_variables = merge(
+    local.dynamic_environment_variables,
+    local.static_env_vars
+  )
+
+  assume_role_policy_statements = {
+    account_root = {
+      effect  = "Allow",
+      actions = ["sts:AssumeRole"],
+      principals = {
+        account_principal = {
+          type        = "Service",
+          identifiers = ["lambda.amazonaws.com"]
+        }
+      }
+    }
+  }
+  attach_policy_statements = true
+  policy_statements = merge(
+    local.policy_statements_password_rotation,
+    local.policy_statements_log_exportation
+  )
+  source_path = "${path.module}/fixtures/functions/${var.which_lambda_function}/lambda_function.py"
```

Comment:
yeah I think we need to account for `which_lambda_function` containing more than 1 function. Either by throwing an error or looping through them in this `main.tf` when calling the upstream module

Reply:
nvm, input var validation handles ensuring that there's an exact match of a single function